### DISCUSSION TITLE
Qual foi seu desafio no processo de entrevista?

### DISCUSSION THREAD

**Discussion by andrecoliveira** [frontendbr/forum#1397] [Discussão]
Olá pessoal, sempre tive o interesse de saber um pouco como está sendo as abordagens das empresas quanto ao processo seletivo para as vagas de front-end.

O intuito desse assunto é saber e também poder compartilhar um pouco as novidades e desafios que estão surgindo nessas etapas de entrevistas. 

Então, se houve uma entrevista técnica, como foi seu desafio?

Falando um pouco do meu último desafio me pediram pra criar um SPA com React, onde eu teria que utilizar a API de mapas do Google e dos Correios. Após incluir um CEP específico, a página deveria apresentar um _pin_ no mapa indicando a rua referente ao CEP digitado.

**Comment by valkervieira** [comment]
A única vez que eu recebi um teste técnico foi há uns 3 anos atrás, era uma vaga de estagiário, tinham 3 níveis de dificuldade.

1. Fazer uma página responsiva com um formulário de login, HTML + CSS + JS simples. Esse eu fiz numa boa.
2. Fazer a mesma página mas como um aplicativo híbrido. Na época não fazia ideia nem de onde começar, ia tentar fazer com Ionic, hoje eu faria com React Native.
3. (Extra) No mesmo aplicativo, depois de logado, abrir a câmera e usar identificação facial para colocar um óculos no rosto. Tirar a foto e marcar a localização. Não fazia ideia mesmo de como fazer, na época eu tava na engenharia e tinha assistido umas apresentações de TCC com processamento de imagem em python, passei a madrugada lendo sobre isso.

O prazo era de 48h, só o item 1 era obrigatório, pedia fosse entregue como um repo aqui no GH com um readme explicativo, eu nem sabia o que era git. Eu não sabia nada na época e nem tive coragem de entregar, hoje eu provavelmente faria numa boa os 3 tópicos dentro do prazo.

Depois disso já fiz mais ou menos umas 10 entrevistas, nunca fui pra parte técnica, sou sempre dispensado depois da conversa inicial ou até na análise do currículo, parece que algo no meu perfil desagrada os entrevistadores. Tenho 3 amigos bem próximos e até alunos que hoje trabalham como desenvolvedores há anos, todos dizem que quando fizeram a entrevista não sabiam nada e mesmo assim foram chamados, não tinham absolutamente nada pra apresentar, um deles inclusive não soube fazer o teste e já está empregado faz 3 anos.

Apesar de eu buscar trabalho na área não tenho esperança de ser contratado, estou me dedicando pra trabalhar por conta e mudar de área, atualmente sou professor, por um lado fico feliz de ver alguns alunos meus se dando bem como programadores, por outro fico bem frustrado de saber que alguns deles mal sabiam mexer no computador quando começaram a estudar aqui na escola e hoje estão em vagas que pagam muito mais do que eu ganho sabendo muito menos do que que eu sei.

Desculpa o desabafo, mas tava falando sobre isso ontem à noite com a minha esposa e aproveitei pra compartilhar.

**Comment by vitor-mendrone-deel** [comment]
Fala @andrecoliveira 

O @felipefialho criou um repositório onde é possível encontrar diversos desafios oficiais de empresas brasileiras e estrangeiras, vale a pena conferir: 

https://github.com/felipefialho/frontend-challenges

**Comment by lucasferreiralimax** [comment]
Eu evito fazer teste e dever de casa para as empresas, ja fui passado para trás muitas vezes.

Hoje em dia se preciso demostrar meu conhecimento técnico deixo previamente avisado que levarei meu notebook para a entrevista e não deixarei nenhum código de presente para a empresa.

Tento manter aplicações  open-source para demostrar o nível de desenvolvimento que estou desenvolvendo no momento, além de participa de comunidades como codepen e aqui também onde posso deixar meus código de abstração front de forma aberta.

Busco também manter meu portfólio atualizado com as últimas tecnologias do mercado que eu estou estudando e dominando no momento.

A realidade do trabalho com tecnologia e intelectual no Brasil ainda é algo tratado com muita falta de respeito e nenhuma empatia com quem trás inovação e soluções.

Vejo muitas empresas ganhando em cima de processos seletivos que elas mesmas mal sabem gerir para avaliar o profissional.

Um assunto muito delicado para se falar levianamente, vejo muitos crimes corporativos com o roubo da mão de obra e trabalho intelectual, nesses processos seletivos.

**Comment by andrecoliveira** [comment]
> A única vez que eu recebi um teste técnico foi há uns 3 anos atrás, era uma vaga de estagiário, tinham 3 níveis de dificuldade.
> 
> 1. Fazer uma página responsiva com um formulário de login, HTML + CSS + JS simples. Esse eu fiz numa boa.
> 2. Fazer a mesma página mas como um aplicativo híbrido. Na época não fazia ideia nem de onde começar, ia tentar fazer com Ionic, hoje eu faria com React Native.
> 3. (Extra) No mesmo aplicativo, depois de logado, abrir a câmera e usar identificação facial para colocar um óculos no rosto. Tirar a foto e marcar a localização. Não fazia ideia mesmo de como fazer, na época eu tava na engenharia e tinha assistido umas apresentações de TCC com processamento de imagem em python, passei a madrugada lendo sobre isso.
> 
> O prazo era de 48h, só o item 1 era obrigatório, pedia fosse entregue como um repo aqui no GH com um readme explicativo, eu nem sabia o que era git. Eu não sabia nada na época e nem tive coragem de entregar, hoje eu provavelmente faria numa boa os 3 tópicos dentro do prazo.
> 
> Depois disso já fiz mais ou menos umas 10 entrevistas, nunca fui pra parte técnica, sou sempre dispensado depois da conversa inicial ou até na análise do currículo, parece que algo no meu perfil desagrada os entrevistadores. Tenho 3 amigos bem próximos e até alunos que hoje trabalham como desenvolvedores há anos, todos dizem que quando fizeram a entrevista não sabiam nada e mesmo assim foram chamados, não tinham absolutamente nada pra apresentar, um deles inclusive não soube fazer o teste e já está empregado faz 3 anos.
> 
> Apesar de eu buscar trabalho na área não tenho esperança de ser contratado, estou me dedicando pra trabalhar por conta e mudar de área, atualmente sou professor, por um lado fico feliz de ver alguns alunos meus se dando bem como programadores, por outro fico bem frustrado de saber que alguns deles mal sabiam mexer no computador quando começaram a estudar aqui na escola e hoje estão em vagas que pagam muito mais do que eu ganho sabendo muito menos do que que eu sei.
> 
> Desculpa o desabafo, mas tava falando sobre isso ontem à noite com a minha esposa e aproveitei pra compartilhar.

Bastante desafiador. Sobre as oportunidades, peço que não desista. O mercado realmente está cada vez mais exigente, passamos por diversas etapas até chegar no processo final. Se hoje você consegue executar as 3 tarefas, mostra que evoluiu bastante. Continue firme.


@Mendrone Nooossa, muito obrigado. Ainda pesquisei sobre o assunto no fórum antes de postar.

@lucasferreiralimax Interessante. Tenho feito muito de forma sreenshare. E toda a equipe acompanhando lá do outro lado.

**Comment by lucasferreiralimax** [comment]
> > A única vez que eu recebi um teste técnico foi há uns 3 anos atrás, era uma vaga de estagiário, tinham 3 níveis de dificuldade.
> > 
> > 1. Fazer uma página responsiva com um formulário de login, HTML + CSS + JS simples. Esse eu fiz numa boa.
> > 2. Fazer a mesma página mas como um aplicativo híbrido. Na época não fazia ideia nem de onde começar, ia tentar fazer com Ionic, hoje eu faria com React Native.
> > 3. (Extra) No mesmo aplicativo, depois de logado, abrir a câmera e usar identificação facial para colocar um óculos no rosto. Tirar a foto e marcar a localização. Não fazia ideia mesmo de como fazer, na época eu tava na engenharia e tinha assistido umas apresentações de TCC com processamento de imagem em python, passei a madrugada lendo sobre isso.
> > 
> > O prazo era de 48h, só o item 1 era obrigatório, pedia fosse entregue como um repo aqui no GH com um readme explicativo, eu nem sabia o que era git. Eu não sabia nada na época e nem tive coragem de entregar, hoje eu provavelmente faria numa boa os 3 tópicos dentro do prazo.
> > Depois disso já fiz mais ou menos umas 10 entrevistas, nunca fui pra parte técnica, sou sempre dispensado depois da conversa inicial ou até na análise do currículo, parece que algo no meu perfil desagrada os entrevistadores. Tenho 3 amigos bem próximos e até alunos que hoje trabalham como desenvolvedores há anos, todos dizem que quando fizeram a entrevista não sabiam nada e mesmo assim foram chamados, não tinham absolutamente nada pra apresentar, um deles inclusive não soube fazer o teste e já está empregado faz 3 anos.
> > Apesar de eu buscar trabalho na área não tenho esperança de ser contratado, estou me dedicando pra trabalhar por conta e mudar de área, atualmente sou professor, por um lado fico feliz de ver alguns alunos meus se dando bem como programadores, por outro fico bem frustrado de saber que alguns deles mal sabiam mexer no computador quando começaram a estudar aqui na escola e hoje estão em vagas que pagam muito mais do que eu ganho sabendo muito menos do que que eu sei.
> > Desculpa o desabafo, mas tava falando sobre isso ontem à noite com a minha esposa e aproveitei pra compartilhar.
> 
> Bastante desafiador. Sobre as oportunidades, peço que não desista. O mercado realmente está cada vez mais exigente, passamos por diversas etapas até chegar no processo final. Se hoje você consegue executar as 3 tarefas, mostra que evoluiu bastante. Continue firme.
> 
> @Mendrone Nooossa, muito obrigado. Ainda pesquisei sobre o assunto no fórum antes de postar.
> 
> @lucasferreiralimax Interessante. Tenho feito muito de forma sreenshare. E toda a equipe acompanhando lá do outro lado.

@andrecoliveira boa dica, mas mesmo assim fica a brecha do outro lado esta gravando a streaming de video e depois so reescrever oque fez. O método notebook olho no olho ainda é o mais seguro na minha opnião, o time deve ser capaz de avaliar tecnicamente aquilo que você fez apenas olhando o projeto. Senão esta trabalhando de graça na entrevista... e isso acontece!

Por isso incentivo projetos open-source para ajudar a sociedade, assim fica subentendido a capacidade técnica disponível para colaboração a ser avaliada.

Minha opnião pessoal e particular como desenvolvedor de soluções tecnologicas.

Compartilhamos o conhecimento de forma aberta na internet para evitar essas ciladas de ser feito de bobo por empresas, isso é uma realidade.

Se for eu para avaliar um teste de algum candidato, eu tenho que ter a capacidade intelectual apenas olhando o código, sem precisar ter ele em mãos de fato. Até para passar mais segurança para o entrevistado pelo o seu esforço e capacidade.

**Comment by anndreyy** [comment]
O meu maior desafio em processos seletivos era sempre ter um feedback, tenho a impressão que há muita falta de respeito com quem é entrevistado.

No fim do ano passado eu resolvi trocar de empresa e participei de três processos onde cada um tinha sua particularidade, mas todos para a mesma vaga: Analista de Sistemas Pleno.

O primeiro era em uma agência de publicidade, onde primeiro teve um teste técnico e na mesma semana uma entrevista com o encarregado, onde só tive uma resposta 2 meses depois porque eu vi ele em um evento e não teve para onde correr, ele me disse que não tinha mais orçamento para aquela contratação por isso não me deu um retorno.
O segundo também foi em uma agência, mas o foco dela era mais a área de dev, fui entrevistado por 5 pessoas ao mesmo tempo e nenhuma era a pessoa que abriu a vaga, porque aparentemente a pessoa não podia participar. Fiquei umas duas semanas cobrando um feedback e nada até hoje.
E o terceiro foi para uma empresa de software na área financeira, onde teve 4 etapas, a primeira com o pessoal do RH, segunda com o gerente do departamento, a terceira com o diretor de ti e por ultimo com o uma moça do DP. Em momento algum eu precisei escrever 1 linha de código. Recebi o feedback do diretor no final da entrevista e é aonde eu estou trabalhando hoje.

**Comment by anndreyy** [comment]
Talvez o meu problema seja com as agências, principalmente com as familiares, sabe aquelas que os donos são todos da mesma família? Trabalhei em uma assim e a minha experiencia foi péssima.

**Comment by felipefialho** [comment]
Linkando essas issues bem legais sobre o assunto #917 #911
